### ISSUE TITLE
Signing issues

### ISSUE THREAD

**Issue by asbjornenge** [derivepass/derivepass-ios#4]
Hi Fedor :smile: 

Was going to give DerivePass a try this evening but got stuck on some signing issues. Not by any means an XCode expert, so I'm probably missing something obvious.

Here is where I get stuck:

<img width="585" alt="screen shot 2017-01-29 at 20 41 20" src="https://cloud.githubusercontent.com/assets/241601/22407299/302c8950-e664-11e6-92d5-92cc31d956c6.png">

I tired using my personal identity, but then I get:

`Your account does not have sufficient permissions to modify containers.` and
`No profiles for 'com.indutny.DerivePass' were found`

What am I missing...? :thinking:

**Comment by indutny** [comment]
Hm... containers...

It sounds like you have to disable CloudKit. Let me push some changes here really quick.

**Comment by indutny** [comment]
Pushed fa77e67, let's try following thing:

1. Comment `DERIVEPASS_HAS_CLOUDKIT` in `ApplicationDataController.h`
2. Open project configuration, `Capabilities` tab and disable CloudKit (as on attached picture)

<img width="824" alt="screen shot 2017-01-29 at 3 01 43 pm" src="https://cloud.githubusercontent.com/assets/238531/22407414/ed19cf86-e633-11e6-8298-caaf9a08cb7f.png">

**Comment by indutny** [comment]
Also, what options do you have for the `Team` select?

**Comment by indutny** [comment]
You would probably need to add your Apple ID account to Xcode settings as well. See: https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppStoreDistributionTutorial/AddingYourAccounttoXcode/AddingYourAccounttoXcode.html

**Comment by asbjornenge** [comment]
Removed the CloudKit business, but getting same error still :-/

I only have 1 option for `Team` which is `Asbjorn Andersen (Personal team)` - which is my Apple ID.

**Comment by indutny** [comment]
Have you tried selecting it? Is at least the container error gone now?

**Comment by asbjornenge** [comment]
It is when I try my personal identity that the `Your account does not have sufficient permissions to modify containers.` shows up.

<img width="613" alt="screen shot 2017-01-29 at 21 15 54" src="https://cloud.githubusercontent.com/assets/241601/22407522/46a2cf88-e668-11e6-8f87-98420bfb50db.png">

Still the same after the CloudKit mod.

**Comment by asbjornenge** [comment]
Quickfix: Add me in TestFlight? :-P

**Comment by indutny** [comment]
We can do this for sure!

It just would be great if we could resolve this issue anyway :wink: 

Have you seen http://stackoverflow.com/questions/40596802/xcode-8-your-account-does-not-have-sufficient-permissions-to-modify-containers already?

**Comment by indutny** [comment]
Also https://github.com/anthonya1999/GoodNight/issues/99

**Comment by indutny** [comment]
One more thing: after you disabled iCloud, what are the contents of `DerivePass/DerivePass/DerivePass.entitlements`?

**Comment by asbjornenge** [comment]
Yeah I saw that stackoverflow one, but was unsure about the meaning of `I then archived my project by manually setting the provisioning profiles`. How do I do that?

I'll take a look at that github thread in a bit, :pizza: time over here :grin: 

```
❯ cat DerivePass/DerivePass/DerivePass.entitlements 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
        <key>com.apple.developer.default-data-protection</key>
        <string>NSFileProtectionComplete</string>
        <key>com.apple.developer.icloud-container-identifiers</key>
        <array>
                <string>iCloud.$(CFBundleIdentifier)</string>
        </array>
        <key>com.apple.developer.icloud-services</key>
        <array>
                <string>CloudKit</string>
        </array>
        <key>com.apple.developer.ubiquity-kvstore-identifier</key>
        <string>$(TeamIdentifierPrefix)$(CFBundleIdentifier)</string>
</dict>
</plist>
```

**Comment by indutny** [comment]
Alright, let's try removing everything under `NSFileProtectionComplete` line, up to the `</dict>` tag.

**Comment by indutny** [comment]
Bon apetit!

**Comment by asbjornenge** [comment]
@indutny thanks! I ate too much and fell asleep :stuck_out_tongue_closed_eyes: After removing the lines from `Derivepass.entitlements` (and restarting XCode a few times), I now get `The 'iCloud' feature is only available to users enrolled in Apple Developer Program. Please visit 
https://developer.apple.com/programs/ to enroll.` which is probably my fault...? Mad meeting monday atm. but I'll get on enrolling later today and post back :smile: :+1:

**Comment by indutny** [comment]
Heya!

It shouldn't really use `iCloud`. That's the point of the changes that we try to do. Enrolling is unfortunately a paid feature, so I hope that it can be avoided in general case.

**Comment by indutny** [comment]
This may help: http://stackoverflow.com/questions/37522049/xcode-free-provisioning-the-apple-push-notification-feature-is-only-available

**Comment by indutny** [comment]
> Delete the certificate from the keychain. Remove your account from the xcode->preference->Accounts. Add your account again. The issue will be resolved. The remote notification feature is not available for free apple developer account.

**Comment by asbjornenge** [comment]
:apple: :bug: 

<img width="341" alt="screen shot 2017-01-30 at 22 40 27" src="https://cloud.githubusercontent.com/assets/241601/22442956/68482d1a-e73d-11e6-937b-ed7e85b8f1e0.png">

Tried the above multiple times, same issue still :-/ Also the `Cloud` option no longer shows up under `Capabilities`. Seems like it must be some config ref. since it fails instantly when I try to build.

Maybe I can try removing and re-adding the whole project in XCode. I'll give that a try later today :+1:

**Comment by indutny** [comment]
Hey hey!

Just got a chance to test it on different laptop. I was able to run it in Simulator after removing everything inside `<dict></dict>` from that entitlements file.

These warnings are still printed, but I don't think that it matters way too much. Could you please give it a try and see if it runs in the Simulator, and hopefully on iPhone as well?

Thank you,
Fedor.

**Comment by asbjornenge** [comment]
Yes, indeed it works in the simulator now! :tada: 

However, when I try to build for phone it still fails, same error :pensive:

**Comment by indutny** [comment]
We're almost there then! :) I'll give it another spin later today.

**Comment by indutny** [comment]
I think I got it working! Please give a try to this patch:

```diff
diff --git a/DerivePass/DerivePass.xcodeproj/project.pbxproj b/DerivePass/DerivePass.xcodeproj/project.pbxproj
index 5acacdf..80848ec 100644
--- a/DerivePass/DerivePass.xcodeproj/project.pbxproj
+++ b/DerivePass/DerivePass.xcodeproj/project.pbxproj
@@ -11,7 +11,6 @@
    4D02DF001E351F7A0076B2F8 /* ValidationPopOverController.m in Sources */ = {isa = PBXBuildFile; fileRef = 4D02DEFF1E351F7A0076B2F8 /* ValidationPopOverController.m */; };
    4D3F05321E326DDF00CA1C2A /* Helpers.m in Sources */ = {isa = PBXBuildFile; fileRef = 4D3F05311E326DDF00CA1C2A /* Helpers.m */; };
    4D3F053C1E3274E200CA1C2A /* DerivePassTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 4D3F053B1E3274E200CA1C2A /* DerivePassTests.m */; };
-		4D6349C31E2BFBD10051D4DF /* CloudKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 4D6349C21E2BFBD10051D4DF /* CloudKit.framework */; };
    4D6349D21E2D218F0051D4DF /* DerivePass.xcdatamodeld in Sources */ = {isa = PBXBuildFile; fileRef = 4D6349D01E2D218F0051D4DF /* DerivePass.xcdatamodeld */; };
    4D6349DB1E2D2E380051D4DF /* TutorialPageViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = 4D6349DA1E2D2E380051D4DF /* TutorialPageViewController.m */; };
    4D6349DE1E2D2E680051D4DF /* TutorialViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = 4D6349DD1E2D2E680051D4DF /* TutorialViewController.m */; };
@@ -143,7 +142,6 @@
      isa = PBXFrameworksBuildPhase;
      buildActionMask = 2147483647;
      files = (
-				4D6349C31E2BFBD10051D4DF /* CloudKit.framework in Frameworks */,
      );
      runOnlyForDeploymentPostprocessing = 0;
    };
@@ -375,18 +373,12 @@
          };
          4D8F97E21E29D9FF0070B586 = {
            CreatedOnToolsVersion = 8.2.1;
-						DevelopmentTeam = VHWE95VV4E;
+						DevelopmentTeam = 94URW7U4W9;
            ProvisioningStyle = Automatic;
            SystemCapabilities = {
-							com.apple.DataProtection = {
-								enabled = 1;
-							};
              com.apple.Push = {
                enabled = 0;
              };
-							com.apple.iCloud = {
-								enabled = 1;
-							};
            };
          };
        };
@@ -691,7 +683,7 @@
        CODE_SIGN_ENTITLEMENTS = DerivePass/DerivePass.entitlements;
        CODE_SIGN_IDENTITY = "iPhone Developer";
        "CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
-				DEVELOPMENT_TEAM = VHWE95VV4E;
+				DEVELOPMENT_TEAM = 94URW7U4W9;
        FRAMEWORK_SEARCH_PATHS = "";
        HEADER_SEARCH_PATHS = (
          ../deps/derivepass,
@@ -701,7 +693,7 @@
        );
        INFOPLIST_FILE = DerivePass/Info.plist;
        LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = com.indutny.DerivePass;
+				PRODUCT_BUNDLE_IDENTIFIER = com.opensource.DerivePass;
        PRODUCT_NAME = "$(TARGET_NAME)";
      };
      name = Debug;
@@ -713,7 +705,7 @@
        ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = "";
        CODE_SIGN_ENTITLEMENTS = DerivePass/DerivePass.entitlements;
        CODE_SIGN_IDENTITY = "iPhone Developer";
-				DEVELOPMENT_TEAM = VHWE95VV4E;
+				DEVELOPMENT_TEAM = 94URW7U4W9;
        FRAMEWORK_SEARCH_PATHS = "";
        HEADER_SEARCH_PATHS = (
          ../deps/derivepass,
@@ -723,7 +715,7 @@
        );
        INFOPLIST_FILE = DerivePass/Info.plist;
        LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = com.indutny.DerivePass;
+				PRODUCT_BUNDLE_IDENTIFIER = com.opensource.DerivePass;
        PRODUCT_NAME = "$(TARGET_NAME)";
      };
      name = Release;
diff --git a/DerivePass/DerivePass/DerivePass.entitlements b/DerivePass/DerivePass/DerivePass.entitlements
index 3729903..6631ffa 100644
--- a/DerivePass/DerivePass/DerivePass.entitlements
+++ b/DerivePass/DerivePass/DerivePass.entitlements
@@ -2,17 +2,5 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>com.apple.developer.default-data-protection</key>
-	<string>NSFileProtectionComplete</string>
-	<key>com.apple.developer.icloud-container-identifiers</key>
-	<array>
-		<string>iCloud.$(CFBundleIdentifier)</string>
-	</array>
-	<key>com.apple.developer.icloud-services</key>
-	<array>
-		<string>CloudKit</string>
-	</array>
-	<key>com.apple.developer.ubiquity-kvstore-identifier</key>
-	<string>$(TeamIdentifierPrefix)$(CFBundleIdentifier)</string>
 </dict>
 </plist>
diff --git a/DerivePass/DerivePass.xcodeproj/project.xcworkspace/xcuserdata/findutnyy.xcuserdatad/UserInterfaceState.xcuserstate b/DerivePass/DerivePass.xcodeproj/project.xcworkspace/xcuserdata/findutnyy.xcuserdatad/UserInterfaceState.xcuserstate
index 8c4eeac..8e30b35 100644
Binary files a/DerivePass/DerivePass.xcodeproj/project.xcworkspace/xcuserdata/findutnyy.xcuserdatad/UserInterfaceState.xcuserstate and b/DerivePass/DerivePass.xcodeproj/project.xcworkspace/xcuserdata/findutnyy.xcuserdatad/UserInterfaceState.xcuserstate differ
diff --git a/DerivePass/DerivePass/ApplicationDataController.h b/DerivePass/DerivePass/ApplicationDataController.h
index d230c8f..f5fbea2 100644
--- a/DerivePass/DerivePass/ApplicationDataController.h
+++ b/DerivePass/DerivePass/ApplicationDataController.h
@@ -11,7 +11,7 @@
 #import <CoreData/CoreData.h>
 #import <Foundation/Foundation.h>
 
-#define DERIVEPASS_HAS_CLOUDKIT 1
+// #define DERIVEPASS_HAS_CLOUDKIT 1
 
 #import "Application+CoreDataProperties.h"
 
```

**Comment by indutny** [comment]
I think that the data protection doesn't really need to be disabled, though, but I haven't tested this scenario separately from other changes.

**Comment by indutny** [comment]
Hey @asbjornenge :wink: any news?

**Comment by asbjornenge** [comment]
Hey mate :smile:

Have a deadline 15 feb. for this RN app I'm working on. Thats cutting it close so I've be bogged down on that. But I have your patch on ***snooze*** for this weekend :wink: I'll let you know!

**Comment by indutny** [comment]
Yikes! Sorry for bothering you then! Good luck with it, and stay focused 😉  👍 !

**Comment by asbjornenge** [comment]
@indutny YES, the patch it works!! :tada: :rocket: 🕺 I guess it was removing these

```
-							com.apple.iCloud = {
-								enabled = 1;
-							};
```

that did the trick. Anyway, when building for the phone - should I use the `Release` scheme? Seems appropriate. Good think to include in any docs about building :smile: :+1: 

Thanks for the help! Now I'll try to use it a bit :grin:

**Comment by indutny** [comment]
Thank you for helping me with this! I'll include it in README tomorrow!

I believe that `Release` should be the most appropriate scheme.

(Let me know if there is anything in the app itself that needs improvement :wink: )

**Comment by indutny** [comment]
Fixed in 5a287df
